pkg/scheduler/algorithm/priorities/resource_allocation.go

```diff
@@ -99,5 +101,17 @@ func getNonZeroRequests(pod *v1.Pod) *schedulernodeinfo.Resource {
 		result.MilliCPU += cpu
 		result.Memory += memory
 	}
+
+	// If Overhead is being utilized, add to the total requests for the pod
```

Comment:
What is the effect of adding the overhead here if the container requests were explicitly set to zero?

Reply:
Interesting - I'm curious the scenario where this would happen.

Either way, if they are running their explicitly-set-zero-request workload in a sandbox (RuntimeClass with Overhead), there is still a cost of running that sandbox, so the result of this call will be just the pod overhead.

To me, this makes sense.  WDYT?

Reply:
Yeah, makes sense, but I think I don't really understand this method. I think maybe it should just be called "getRequestsSum" - I think non-zero just means apply defaulting?

Reply:
Correct -- for accounting purposes it adds a non-zero value for requests that were zero.  The naming and utilization wasn't very clear to me at first.  In either case, though, the Overhead addition is needed.  There was another PR opened to help address the naming/usage: see https://github.com/kubernetes/kubernetes/pull/78263

Reply:
- `requestedResources` is the exact sum of pod resource requests on that node.
- `nonzeroRequest` is a modified version where we apply a minimum value to CPU and memory for any container that doesn't specify a value.